theatre/studio/src/panels/SequenceEditorPanel/RightOverlay/Markers.tsx

```diff
@@ -0,0 +1,36 @@
+import type {Pointer} from '@theatre/dataverse'
+import {useVal} from '@theatre/react'
+import getStudio from '@theatre/studio/getStudio'
+import React from 'react'
+import styled from 'styled-components'
+import type {SequenceEditorPanelLayout} from '@theatre/studio/panels/SequenceEditorPanel/layout/layout'
+import Marker from './Marker'
+
+const Container = styled.div`
+  position: absolute;
+  top: 15px;
+  left: 0;
+  width: 100%;
+  height: 40px;
+  overflow: hidden;
+`
+
+const Markers: React.VFC<{layoutP: Pointer<SequenceEditorPanelLayout>}> = ({
+  layoutP,
+}) => {
+  const sheetAddress = useVal(layoutP).sheet.address
+  const markers = useVal(getStudio()!.atomP)?.historic.projects
```

Comment:
These two lines are also calling `useVal()` with a close-to-root pointer.

Suggestion:

```
const sheet = useVal(layoutP.sheet)
const markers = useVal(getStudio()!.atomP.historic.projects.stateByProjectId[sheet.address.projectId].stateBySheetId[sheetAddress.sheetId].sequenceEditor.markers)
```

Good thing is, you won't even need to do all the optional chaining (`?.`) because the return type of `useVal()` will be `T | undefined` if there are any `undefined`s in the path.